WalletWasabi/Tor/Http/TorHttpClient.cs

```diff
@@ -178,22 +178,25 @@ public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request,Canc
 			request.Version = HttpProtocol.HTTP11.Version;
 			request.Headers.AcceptEncoding.Add(new StringWithQualityHeaderValue("gzip"));
 
-			if (TorSocks5Client is { } && !TorSocks5Client.IsConnected)
+			if (TorSocks5Client is null)
```

Comment:
The old code has some kind of recovery mechanism in case of `!TcpClient.Connected`. Now,` TorSocks5Client = null;` is required to have a new handshake. 

So if something happens after line 201 where TorSocks5Client will be set to null?

Reply:
So `TorSocks5Client = null` at 183 and in `try` section either we succeed in setting `TorSocks5Client` to a non-null value or we throw at 199. 

So you get to line 201 and on only when when `TorSocks5Client is NOT null`.

The recovery mechanism you talked about is this: https://github.com/kiminuo/WalletWasabi/blob/feature/2021-02-21-TorSocks5Client-connected/WalletWasabi/Tor/Http/TorHttpClient.cs#L90-L138 I think.

Reply:
In the old behavior, `TorSocks5Client` was not set explicitely to null and `Connected` property was checked because a TCP connection might have not been established in a previous try.

Reply:
It's a little bit unintuitive but I think the new behavior is correct.

Reply:
Maybe a few lines of comment can help, otherwise Concept ACK.

Reply:
Where and why can it becomes null.

Reply:
Is it clearer now? I have removed a few lines that are unnecessary now.

Reply:
Ok, now I think it is very clear when it can be `null`.